### ISSUE TITLE
Vehicles block bullets without taking damage (build 3604)

### ISSUE THREAD

**Issue by ejseto** [CleverRaven/Cataclysm-DDA#13392]
Spawn a motorcycle, place a turret on one side, stand on the other and fire at the turret across the front wheel. Neither the turret nor the motorcycle will take damage.

**Comment by Coolthulhu** [comment]
Vehicles are supposed not to have self-damage from non-AoE attacks.

**Comment by drbig** [comment]
@Coolthulhu  "non-AoE" expand please?

> Spawn a motorcycle, place a turret on one side, stand on the other and fire at the turret across the front wheel. Neither the turret nor the motorcycle will take damage.

That was a major bug for some time.

**Comment by Coolthulhu** [comment]
> @Coolthulhu "non-AoE" expand please?

Bullets. A vehicle shouldn't wreck itself by shooting at stuff. Turrets are mounted on the roof, so it doesn't make sense for them to hit boards on the way.

**Comment by drbig** [comment]
I meant what "AoE" means exactly (Alterans over Else? Age of Empires?).

**Comment by Coolthulhu** [comment]
Area of Effect
One of the most common acronyms out there

**Comment by drbig** [comment]
Noted down, no more question (...though i wouldn't capitalise it like that ;))

**Comment by ejseto** [comment]
No, you've misunderstood. Firing (a hand weapon, not a vehicle turret) at a target (I used a turret because it is both place-able and stationary) with a vehicle in the direct path of the bullet causes the bullet to disappear and cause no damage to either vehicle or target if the vehicle tile in the path is a motorcycle wheel.

**Comment by ejseto** [comment]
Ok, after reading through the code, I think the problem is that someone misunderstood the purpose of the return value of vehicle::damage().

```
    // Damage individual part. bash means damage
    // must exceed certain threshold to be substracted from hp
    // (a lot light collisions will not destroy parts)
    // Returns damage bypassed
    int damage (int p, int dmg, damage_type type = DT_BASH, bool aimed = true);
```

The "damage bypassed" would be 100% for parts that shouldn't block bullets, like wheels and frames, should it not? However in map::shoot() this value is subtracted from the projectile damage, causing bullets that should pass over wheels to instead lose all their damage and thus be stopped. It seems someone misunderstood it as the damage *caused*, not the damage *remaining*.

In map::shoot(), replacing the following "-=" operator with "=" fixes this behavior, allowing bullets to pass over wheels and frames and penetrate multiple tiles through 0-hp greyed-out wrecks, scattering their debris.

```
if( veh != nullptr ) {
        const bool inc = (ammo_effects.count("INCENDIARY") || ammo_effects.count("FLAME"));
        dam -= veh->damage( vpart, dam, inc ? DT_HEAT : DT_BASH, hit_items );
    }
```

**Comment by BevapDin** [comment]
I say this has been fixed by #13710, if anybody has any objections speak now or forever hold your peace (-:
